cabot/plugin_test_utils.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+from django.utils import timezone
+from django.contrib.auth.models import Permission, User
+from rest_framework.test import APITestCase
+
+from cabot.cabotapp.alert import update_alert_plugins
+from cabot.cabotapp.models import (
+    JenkinsStatusCheck,
+    HttpStatusCheck,
+    TCPStatusCheck,
+    Service,
+    StatusCheckResult,
+    Schedule,
+    Shift,
+    get_duty_officers,
+    get_fallback_officers,
+    UserProfile)
+
+
+class PluginTestCase(APITestCase):
+    """
+    Sets up the following testing environment:
```

Comment:
I would start this comment with a high-level explanation of what this class does and what it provides.  Eg:

- This class provides a common testing environment and utility functions for Cabot test cases.
- The common environment includes users, schedules, checks, and a service (see below).
- Includes a utility function for setting test states (`set_test_states`) and service statuses (`set_service_status`).

.... and then continue with the existing comments, which provide details.